### ISSUE TITLE
new functionalities

### ISSUE THREAD

**Issue by hurielreichel** [hurielreichel/tidyopeneo#4]
- [x] 1. pass .p and .con as function parameters as in Edzer's commit.
- [x] 2. group_by + summarise behaviour {if data cude is grouped, or not, different interactions} - add new subclass for that and therefore a (sub)method. 
- [x] 3. pass what slice does to filter. 
- [x] 4. year %in% (2000:2002) / group_by(year, month) [use list_connection()$extent as reference
- [x] 5. check mandatory params -> is.null() stop()
- [x] 6. "datacube" should be the first in the list of inherits(), not last
- [ ] 7. Maths and Ops Math.datacube = function(x, ...) { p = openeo::processes(); p[[.Generic]](x) }
- [x] 8. Add examples for all functions + add them to the vignette
- [x] 9. create slice interaction more similar dplyr slice 
- [x] 10. remove default con from all function... speed up code.

**Comment by edzer** [comment]
Yes; see also https://github.com/Open-EO/openeo-r-client/issues/141

**Comment by hurielreichel** [comment]
Thanks a lot @edzer ! Working on it!

**Comment by hurielreichel** [comment]
@edzer for number 4, the filter using very dplyrish verbs... 

the only way I see this working is to store somehow time related (maybe even space related) objects/columns in the class "datacube". I'd then add "year", "month", "day", "x", "y", create many `filter_temporal` and `filter_spatial` and finally use a `merge_cubes()` process. 

What do you think of this idea?

**Comment by hurielreichel** [comment]
I'll discuss it with matthias, see if there's a way to do filter_temporal with length > 2 time_ext lists.

**Comment by hurielreichel** [comment]
@edzer could you have a look at https://github.com/hurielreichel/tidyopeneo/blob/main/R/Math.R
It's mainly your code, so it would help if you can check it. 
Thanks :)
